control/netns_utils.go

```diff
@@ -0,0 +1,303 @@
+package control
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"runtime"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	NsName       = "daens"
+	HostVethName = "dae0"
+	NsVethName   = "dae0peer"
+)
+
+var (
+	daeNetns *DaeNetns
+)
+
+type DaeNetns struct {
```

Comment:
Do we need to implement `func Close() error` ?

Reply:
Who calls this Close()?

Reply:
> Who calls this Close()?

We should make sure that daens is within the program lifecycle or the control plane lifecycle. I think it should be within the control plane lifecycle to avoid data interrupt when reloading.

Reply:
> Thought about the lifecycle idea but gave it up, because `sendPkt` is an independent function without access to *ControlPlane.

That is to say, we can add `defer daeNs.Close()` at: https://github.com/daeuniverse/dae/blob/8beaa5bbe72a968edd119dd5f095b7600737af0a/cmd/run.go#L115